pkg/api/norman/customization/cluster/validator.go

```diff
@@ -329,13 +329,16 @@ func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec
 
 }
 
-func (v *Validator) validateEKSConfig(request *types.APIContext, spec *v32.ClusterSpec) error {
-	if spec.EKSConfig == nil {
+func (v *Validator) validateEKSConfig(request *types.APIContext, cluster map[string]interface{}) error {
+	eksConfig, ok := cluster["eksConfig"].(map[string]interface{})
+	if !ok {
 		return nil
 	}
 
+	amazonCredential, _ := eksConfig["amazonCredentialSecret"].(string)
```

Comment:
Don't want to return an error if the secret is empty ?

Reply:
we could although an error should be returned by access.ByID